app/src/components/OrderBookSnapshot.js

```diff
@@ -90,103 +109,161 @@ class OrderBookSnapshot extends Component {
-                                className={classes.timestampSlider}
-                                min={NANOSECONDS_IN_NINE_AND_A_HALF_HOURS} // nanoseconds between 12am and 9:30am
-                                max={NANOSECONDS_IN_SIXTEEN_HOURS} // nanoseconds between 12am and 4pm
-                                step={1}
-                                defaultValue={0}
-                                onChange={this.handleChangeTime}
-                                marks={[
-                                    {
-                                        value: NANOSECONDS_IN_NINE_AND_A_HALF_HOURS,
-                                        label: '9:30AM',
-                                    },
-                                    {
-                                        value: NANOSECONDS_IN_SIXTEEN_HOURS,
-                                        label: '4:00PM',
-                                    },
-                                ]}
-                                onChangeCommitted={this.handleCommitTime}
-                            />
-                        </FormControl>
-                    </div>
-                </Collapse>
-                <Card>
-                    <TimestampOrderBookScroller
-                        orderBook={{ asks, bids }}
-                    />
-                </Card>
-            </Typography>
-        );
-    }
-}
-
-export default withStyles(Styles)(OrderBookSnapshot);
\ No newline at end of file
+    };
+
+    /**
+     * @desc Gets the next immediate message for the current timestamp, and updates the orderbook with the message's timestamp
+     */
+    handleGoToNextMessage = () => {
+        this.handleGoToMessageByOffset(1);
+    };
+
+    /**
+     * @desc Gets the previous immediate message for the current timestamp, and updates the orderbook with the message's timestamp
+     */
+    handleGoToPreviousMessage = () => {
+        this.handleGoToMessageByOffset(-1);
+    };
+
+    /**
+     * @desc Gets the message for the given offset and updates the order book with the message's timestamp
+     *
+     * @param offset The number of messages to skip forward or backward to
+     */
+    handleGoToMessageByOffset = (offset) => {
+        let { selectedDateTimeNano } = this.state;
+        OrderBookService.getOrderBookPricesByMessageOffset(SNAPSHOT_INSTRUMENT, selectedDateTimeNano, offset)
+            .then(response => {
+                let { timeNanoseconds, dateNanoseconds } = splitNanosecondEpochTimestamp(response.data.timestamp);
+                this.setState({
+                    selectedDateNano: dateNanoseconds,
+                    selectedDateString: epochToDateString(dateNanoseconds),
+                    selectedTimeNano: timeNanoseconds,
+                    selectedTimeString: nanosecondsToString(timeNanoseconds),
+                    selectedDateTimeNano: response.data.timestamp,
+                    asks: response.data.asks,
+                    bids: response.data.bids,
```

Comment:
This won't work with the backend implementation, as we are not sending a full orderbook back. Rather the asks and bids should be "upserted" into the current orderbook state